cmd/describe.go

```diff
@@ -104,3 +108,111 @@ func bodyAsJson(data []byte) ([]byte, error) {
 	}
 	return j.Bytes(), nil
 }
+
+type BaseDescribe struct {
+	raw        []byte
+	kind       jwt.ClaimType
+	outputFile string
+}
+
+func (p *BaseDescribe) Init() error {
+	token, err := jwt.ParseDecoratedJWT(p.raw)
+	if err != nil {
+		return err
+	}
+	p.raw = []byte(token)
+	gc, err := jwt.DecodeGeneric(token)
+	if err != nil {
+		return err
+	}
+	p.kind = gc.ClaimType()
+	return nil
+}
+
+func (p *BaseDescribe) Describe(ctx ActionCtx) (store.Status, error) {
+	var out []byte
+	var err error
+
+	if Raw {
+		out, err = p.Raw(!IsStdOut(p.outputFile))
+	} else if Json || JsonPath != "" {
+		out, err = p.JSON(JsonPath)
+	} else {
+		out, err = p.Structured()
+	}
+	if err != nil {
+		return nil, err
+	}
+	if IsStdOut(p.outputFile) {
+		_, err = fmt.Fprintln(ctx.CurrentCmd().OutOrStdout(), string(out))
+	} else {
+		err = WriteFile(p.outputFile, out)
+	}
+	if err != nil {
+		return nil, err
+	}
+	if !IsStdOut(p.outputFile) {
+		k := "description"
+		if Raw {
+			k = "jwt"
+		}
+		return store.OKStatus("wrote %s %s to %#q", string(p.kind), k, AbbrevHomePaths(p.outputFile)), nil
+	}
+	return nil, err
```

Comment:
Same here about returning `nil, nil` due to the pre-check on `err != nil`.

Reply:
correct